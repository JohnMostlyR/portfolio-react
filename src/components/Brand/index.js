import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {rem} from 'polished';

import Media from '../MediaObject/index';

import MyLogo from '../../images/my-logo.svg';
import typography from '../../styles/templates/typography';

const Wrapper = styled.div`
  float: left;
  padding: 1vh 0 0 1vw;
`;

export const BrandLink = styled.a.attrs({
  href: props => props.href || '#',
})`
  display: inline-block;
  width: ${rem('45px')};
  color: #fff;
  text-decoration: none;
  
  ${typography.minion}
`;

const Brand = ({href, imageSource, text}) => (
  <Wrapper>
    <BrandLink href={href}>
      <Media
        bodyAlign='middle'
        imageSource={imageSource}
        imageAlt='logo'
        imageAlign='middle'
        imageHeight='45px'
        imageWidth='45px'
      >{text}</Media>
    </BrandLink>
  </Wrapper>
);

Brand.propTypes = {
  href: PropTypes.string,
  imageSource: PropTypes.string,
  text: PropTypes.string,
};

Brand.defaultProps = {
  href: '/',
  imageSource: MyLogo,
  text: 'JOHAN MEESTER',
};

export default Brand;
