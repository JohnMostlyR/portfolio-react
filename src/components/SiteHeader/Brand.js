import React from 'react';
import styled from 'styled-components';
import Media from '../MediaObject/Media';
import typography from '../../helpers/styles/templates/typography';

import MyLogo from '../../images/my-logo.svg';

const BrandWrapper = styled.div`
  float: left;
  padding: 1vh 0 0 1vw;
`;

const BrandLink = styled.a.attrs({
  href: props => props.href || '#',
})`
  display: inline-block;
  width: 45px;
  color: #fff;
  text-decoration: none;
  
  ${typography.minion}
`;

const Brand = (props) => (
  <BrandWrapper>
    <BrandLink href={'/intro'}>
      <Media
        bodyAlign='middle'
        imageSource={MyLogo}
        imageAlt='logo'
        imageAlign='middle'
        imageHeight='45px'
        imageWidth='45px'
      >JOHAN MEESTER</Media>
    </BrandLink>
  </BrandWrapper>
);

export default Brand;
