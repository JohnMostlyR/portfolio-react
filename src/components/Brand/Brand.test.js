import React from 'react';
import {shallow} from 'enzyme';

import Brand, {BrandLink} from './index';

import Media from '../MediaObject/index';

describe('Brand', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Brand/>);
  });

  it('should render and match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a `BrandLink` component', () => {
    expect(wrapper.find(BrandLink)).toHaveLength(1);
  });

  it('that should have a default `href` property', () => {
    expect(wrapper.find(BrandLink).prop('href')).toBe('/');
  });

  it('that should receive the `href` property', () => {
    const link = `/${Date.now()}`;
    wrapper.setProps({href: link});
    expect(wrapper.find(BrandLink).prop('href')).toBe(link);
  });

  it('should have a `Media` component', () => {
    expect(wrapper.find(Media)).toHaveLength(1);
  });

  it('that should have a default `imageSource` property', () => {
    expect(wrapper.find(Media).prop('imageSource')).toBeTruthy();
  });

  it('that should receive the `imageSource` property', () => {
    const imageSource = `/${Date.now()}.png`;
    wrapper.setProps({imageSource});
    expect(wrapper.find(Media).prop('imageSource')).toBe(imageSource);
  });

  it('that should have a default `text` property', () => {
    expect(wrapper.find(Media).children()).toHaveLength(1);
  });

  it('that should receive the `text` property', () => {
    const text = Date.now().toString();
    wrapper.setProps({text});
    expect(wrapper.find(Media).childAt(0).text()).toBe(text);
  });
});