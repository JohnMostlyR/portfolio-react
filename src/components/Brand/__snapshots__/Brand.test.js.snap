// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand should render and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Brand
    href="/"
    imageSource="my-logo.svg"
    text="JOHAN MEESTER"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <styled.a
        href="/"
>
        <Media
                bodyAlign="middle"
                imageAlign="middle"
                imageAlt="logo"
                imageHeight="45px"
                imageSource="my-logo.svg"
                imageWidth="45px"
                spacing="0"
        >
                JOHAN MEESTER
        </Media>
</styled.a>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Media
          bodyAlign="middle"
          imageAlign="middle"
          imageAlt="logo"
          imageHeight="45px"
          imageSource="my-logo.svg"
          imageWidth="45px"
          spacing="0"
>
          JOHAN MEESTER
</Media>,
        "href": "/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "bodyAlign": "middle",
          "children": "JOHAN MEESTER",
          "imageAlign": "middle",
          "imageAlt": "logo",
          "imageHeight": "45px",
          "imageSource": "my-logo.svg",
          "imageWidth": "45px",
          "spacing": "0",
        },
        "ref": null,
        "rendered": "JOHAN MEESTER",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.a
          href="/"
>
          <Media
                    bodyAlign="middle"
                    imageAlign="middle"
                    imageAlt="logo"
                    imageHeight="45px"
                    imageSource="my-logo.svg"
                    imageWidth="45px"
                    spacing="0"
          >
                    JOHAN MEESTER
          </Media>
</styled.a>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Media
            bodyAlign="middle"
            imageAlign="middle"
            imageAlt="logo"
            imageHeight="45px"
            imageSource="my-logo.svg"
            imageWidth="45px"
            spacing="0"
>
            JOHAN MEESTER
</Media>,
          "href": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "bodyAlign": "middle",
            "children": "JOHAN MEESTER",
            "imageAlign": "middle",
            "imageAlt": "logo",
            "imageHeight": "45px",
            "imageSource": "my-logo.svg",
            "imageWidth": "45px",
            "spacing": "0",
          },
          "ref": null,
          "rendered": "JOHAN MEESTER",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
